src/components/ComponentSettingsModal.jsx

```diff
@@ -0,0 +1,467 @@
+          fileUrl: '',
+        },
+      });
+    }
+  }
+
+  deleteHandler = async () => {
+    try {
+      const { sha, baseApiUrl } = this.state;
+      const { fileName } = this.props;
+      const params = { sha };
+      await axios.delete(`${baseApiUrl}/pages/${fileName}`, {
+        data: params,
+      });
+
+      // Refresh page
+      window.location.reload();
+    } catch (err) {
+      console.log(err);
+    }
+  }
+
+  saveHandler = async () => {
+    try {
+      const {
+        title, permalink, fileUrl, date, mdBody, sha, category, prevCategory, baseApiUrl, type, thirdNavTitle,
+      } = this.state;
+      const { fileName, isNewFile, siteName } = this.props;
+      
+      let newFileName, frontMatter
+      if (type === "resource") {
+        newFileName = generateResourceFileName(title, date);
+        frontMatter = { title: enquoteString(title), date };
+      } else if (type === "page") {
+        frontMatter = thirdNavTitle
+          ? { title, permalink, third_nav_title: thirdNavTitle }
+          : { title, permalink };
+        if (category) {
+          const groupIdentifier = fileName.split('-')[0];
+          newFileName = generateCollectionPageFileName(title, groupIdentifier);
+        } else {
+          newFileName = generatePageFileName(title);
+        }
+      }
+
+      if (permalink) {
+        frontMatter.permalink = `/${category ? `${category}/` : ''}${permalink}`;
+      }
+      if (fileUrl) {
+        frontMatter.file_url = fileUrl;
+      }
+      let newBaseApiUrl
+      if (prevCategory) {
+        // baseApiUrl can be used as is because user cannot change categories
+        newBaseApiUrl = baseApiUrl
+      } else {
+        if (category) {
+          // User is adding file to category from main page
+          newBaseApiUrl = `${baseApiUrl}/${type === "resource" ? `resources/${category}` : `collections/${category}`}`
+        } else {
+          // User is adding file with no collections, only occurs for pages
+          newBaseApiUrl = baseApiUrl
+        }
+      }
+
+      const content = concatFrontMatterMdBody(frontMatter, mdBody);
+      const base64EncodedContent = Base64.encode(content);
+
+      let params = {};
+      if (newFileName !== fileName || prevCategory !== category) {
+        // We'll need to create a new .md file with a new filename
+        params = {
+          content: base64EncodedContent,
+          pageName: newFileName,
+        };
+
+        // If it is an existing file, delete the existing page
+        if (!isNewFile) {
+          await axios.delete(`${newBaseApiUrl}/pages/${fileName}`, {
+            data: {
+              sha,
+            },
+          });
+        }
+        await axios.post(`${newBaseApiUrl}/pages`, params);
+      } else {
+        // Save to existing .md file
+        params = {
+          content: base64EncodedContent,
+          sha,
+        };
+        await axios.post(`${newBaseApiUrl}/pages/${fileName}`, params);
+      }
```

Comment:
Since none of these involve state, can we put this in a a separate util function and import it since the file is getting really long?

Reply:
Resolved in https://github.com/isomerpages/isomercms-frontend/pull/184/commits/d7d52bfa188eb098568a0b0165fe181f903451e1